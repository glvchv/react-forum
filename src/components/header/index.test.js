import React from 'react';
import renderer from 'react-test-renderer';
import Header from '.';
import { BrowserRouter } from 'react-router-dom';
import AuthContext from '../../context/authContext';

jest.mock('../button', () => 'Button');


describe('Header component', () => {
    it('should render guest links', () => {
        const tree = renderer.create(
            <BrowserRouter>
                <Header />
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render auth links', () => {
        localStorage.setItem('userId', 123)
        const tree = renderer.create(
            <AuthContext.Provider value={{
                user: {
                    loggedIn: true
                }
            }}>
                <BrowserRouter>
                    <Header />
                </BrowserRouter>

            </AuthContext.Provider>
        ).toJSON();
    })
})