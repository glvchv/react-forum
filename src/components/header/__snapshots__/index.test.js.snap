// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render guest links 1`] = `
<header>
  <a
    className="forum"
    href="/"
    onClick={[Function]}
  >
    &lt; Forum /&gt;
  </a>
  <div
    className="links"
  >
    <a
      href="/rules"
      onClick={[Function]}
    >
      Rules
    </a>
  </div>
  <Button
    link="/login"
    text="Login"
    type="default"
  />
</header>
`;
